bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcarePetHandler.java

```diff
@@ -0,0 +1,251 @@
+                                                new StringType(pet.status.activity.where.toString()));
+                                        updateState(PET_CHANNEL_LOCATION_CHANGED,
+                                                new DateTimeType(pet.status.activity.getLocationChanged()));
+                                    } catch (NumberFormatException e) {
+                                        logger.warn("Invalid location id: {}, ignoring command", newLocationIdStr);
+                                    } catch (SurePetcareApiException e) {
+                                        logger.warn("Error from SurePetcare API. Can't update location {} for pet {}",
+                                                newLocationIdStr, pet);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    break;
+                case PET_CHANNEL_LOCATION_TIMEOFFSET:
+                    logger.debug("Received location time offset update command: {}", command.toFullString());
+                    if (command instanceof StringType) {
+                        synchronized (petcareAPI) {
+                            SurePetcarePet pet = petcareAPI.getPet(thing.getUID().getId());
+                            if (pet != null) {
+                                String commandIdStr = ((StringType) command).toString();
+                                try {
+                                    Integer commandId = Integer.valueOf(commandIdStr);
+                                    Integer currentLocation = pet.status.activity.where;
+                                    logger.debug("Received new location: {}", currentLocation == 1 ? 2 : 1);
+                                    // We set the location to the opposite state.
+                                    // We also set location to INSIDE (1) if currentLocation is Unknown (0)
+                                    if (commandId == 10 || commandId == 30 || commandId == 60) {
+                                        petcareAPI.setPetLocation(pet, currentLocation == 1 ? 2 : 1, new Date(
+                                                System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(commandId)));
+                                    }
+                                    updateState(PET_CHANNEL_LOCATION,
+                                            new StringType(pet.status.activity.where.toString()));
+                                    updateState(PET_CHANNEL_LOCATION_CHANGED,
+                                            new DateTimeType(pet.status.activity.getLocationChanged()));
+                                    updateState(PET_CHANNEL_LOCATION_TIMEOFFSET, UnDefType.UNDEF);
+                                } catch (NumberFormatException e) {
+                                    logger.warn("Invalid location id: {}, ignoring command", commandIdStr);
```

Comment:
This should be a debug log I suppose

Reply:
Please apply this also to the methods above.

Reply:
The location change function of this binding is the most used one... and we'd like to keep this at warn level... so the user can instantly see whats the problem without setting log level to debug...
If you still want me to change this i will do. just let me know.

Reply:
Okay, let's leave this one but maybe add some more context if possible/applicable. Did you check other `warn`-logs for their importance?